### PR TITLE
Fold CommandLineOptions and WireCompiler.

### PR DESCRIPTION
**Pull request by swankjesse**
Now that WireCompiler lets other classes do most of the heavy lifting,
its convenient to define options near to where they're used.

### CODE REVIEW

File: wire-compiler/src/main/java/com/squareup/wire/WireCompiler.java [hunk 1]
```diff
@@ -20,32 +20,120 @@
...
+  final List<String> protoPaths;
+  final String javaOut;
+  final List<String> sourceFileNames;
+  final IdentifierSet identifierSet;
+  final boolean emitOptions;
+  final boolean dryRun;
+  final boolean emitAndroid;
+  final boolean emitCompact;
+
+  public WireCompiler(FileSystem fs, WireLogger log, List<String> protoPaths, String javaOut,
```
Comment (JakeWharton) [inline wire-compiler/src/main/java/com/squareup/wire/WireCompiler.java]: private or package scope this guy? historically it's changed over the course of a major version when new major features come in.

File: wire-compiler/src/main/java/com/squareup/wire/WireCompiler.java [hunk 2]
```diff
@@ -53,43 +141,98 @@ public static void main(String... args) throws IOException {
     }
   }
 
-  static WireCompiler create(CommandLineOptions options) throws WireException {
-    return new WireCompiler(options, FileSystems.getDefault(),
-        new ConsoleWireLogger(options.quiet));
+  static WireCompiler forArgs(String... args) throws WireException {
```
Comment (JakeWharton) [inline wire-compiler/src/main/java/com/squareup/wire/WireCompiler.java]: and `public` here instead

File: wire-compiler/src/main/java/com/squareup/wire/WireCompiler.java (hunk 1 above)
Comment (swankjesse) [inline wire-compiler/src/main/java/com/squareup/wire/WireCompiler.java]: Definitely package-private. The only API this thing offers is `main()`.

File: wire-compiler/src/test/java/com/squareup/wire/CommandLineOptionsTest.java [hunk 3]
```diff
@@ -1,11 +1,9 @@
 package com.squareup.wire;
```
Comment (JakeWharton) [inline wire-compiler/src/test/java/com/squareup/wire/CommandLineOptionsTest.java]: Class name should be renamed

Comment (JakeWharton) [inline wire-compiler/src/test/java/com/squareup/wire/CommandLineOptionsTest.java]: Oh there already is a `WireCompilerTest`

**Comment by JakeWharton** [comment]
LGTM!

**Comment by swankjesse** [comment]
Fixed so only `main` is public. Making anything else public isn't that useful, the only thing you can do with a `WireCompiler` is what main offers.

This class mostly just serves as a model for plugin authors.

Also, worth considering switching this to use one of the JCommander or something.

**Pull request merged**